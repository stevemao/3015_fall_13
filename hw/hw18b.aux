\relax 
\@writefile{toc}{\contentsline {section}{\numberline {1}Type Inference}{1}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1.1}Proof Rules}{1}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1.2}Implementation}{3}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1.3}Adding product types.}{3}}
