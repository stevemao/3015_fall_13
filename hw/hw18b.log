This is pdfeTeX, Version 3.141592-1.21a-2.2 (Web2C 7.5.4) (format=pdflatex 2008.2.21)  18 NOV 2008 13:56
entering extended mode
**hw18b
(./hw18b.tex
LaTeX2e <2003/12/01>
Babel <v3.8d> and hyphenation patterns for american, french, german, ngerman, b
ahasa, basque, bulgarian, catalan, croatian, czech, danish, dutch, esperanto, e
stonian, finnish, greek, icelandic, irish, italian, latin, magyar, norsk, polis
h, portuges, romanian, russian, serbian, slovak, slovene, spanish, swedish, tur
kish, ukrainian, nohyphenation, loaded.
(/usr/share/texmf/tex/latex/base/article.cls
Document Class: article 2004/02/16 v1.4f Standard LaTeX document class
(/usr/share/texmf/tex/latex/base/size11.clo
File: size11.clo 2004/02/16 v1.4f Standard LaTeX file (size option)
)
\c@part=\count79
\c@section=\count80
\c@subsection=\count81
\c@subsubsection=\count82
\c@paragraph=\count83
\c@subparagraph=\count84
\c@figure=\count85
\c@table=\count86
\abovecaptionskip=\skip41
\belowcaptionskip=\skip42
\bibindent=\dimen102
)
\@input{header.aux}
\openout2 = `header.aux'.

(./header.tex (/usr/share/texmf/tex/latex/amsfonts/amssymb.sty
Package: amssymb 2002/01/22 v2.2d

(/usr/share/texmf/tex/latex/amsfonts/amsfonts.sty
Package: amsfonts 2001/10/25 v2.2f
\@emptytoks=\toks14
\symAMSa=\mathgroup4
\symAMSb=\mathgroup5
LaTeX Font Info:    Overwriting math alphabet `\mathfrak' in version `bold'
(Font)                  U/euf/m/n --> U/euf/b/n on input line 132.
)) (./remark.sty)
(/home/faculty/jlc/papers/tex-stuff/bussproofs.sty
Proof Tree (bussproofs) style macros. Version 0.6c.
\theLevel=\count87
\myMaxLevel=\count88
\myBoxA=\box26
\myBoxB=\box27
\myBoxC=\box28
\myBoxD=\box29
\myBoxLL=\box30
\myBoxRL=\box31
\thisAboveSkip=\dimen103
\thisBelowSkip=\dimen104
\newScoreStart=\dimen105
\newScoreEnd=\dimen106
\newCenter=\dimen107
\displace=\dimen108
\leftLowerAmt=\dimen109
\rightLowerAmt=\dimen110
\scoreHeight=\dimen111
\scoreDepth=\dimen112
)
\c@theorem=\count89
\c@conjecture=\count90
\c@corollary=\count91
\c@example=\count92
\c@fact=\count93
\c@lemma=\count94
\c@definition=\count95
\c@claim=\count96
\c@remark=\count97
\c@slogan=\count98
\c@axiom=\count99
\c@note=\count100
\c@problem=\count101
\c@exercise=\count102
) (./hw18b.aux)
\openout1 = `hw18b.aux'.

LaTeX Font Info:    Checking defaults for OML/cmm/m/it on input line 5.
LaTeX Font Info:    ... okay on input line 5.
LaTeX Font Info:    Checking defaults for T1/cmr/m/n on input line 5.
LaTeX Font Info:    ... okay on input line 5.
LaTeX Font Info:    Checking defaults for OT1/cmr/m/n on input line 5.
LaTeX Font Info:    ... okay on input line 5.
LaTeX Font Info:    Checking defaults for OMS/cmsy/m/n on input line 5.
LaTeX Font Info:    ... okay on input line 5.
LaTeX Font Info:    Checking defaults for OMX/cmex/m/n on input line 5.
LaTeX Font Info:    ... okay on input line 5.
LaTeX Font Info:    Checking defaults for U/cmr/m/n on input line 5.
LaTeX Font Info:    ... okay on input line 5.

Underfull \hbox (badness 10000) in paragraph at lines 6--7

 []

LaTeX Font Info:    Try loading font information for U+msa on input line 28.
(/usr/share/texmf/tex/latex/amsfonts/umsa.fd
File: umsa.fd 2002/01/19 v2.2g AMS font definitions
)
LaTeX Font Info:    Try loading font information for U+msb on input line 28.

(/usr/share/texmf/tex/latex/amsfonts/umsb.fd
File: umsb.fd 2002/01/19 v2.2g AMS font definitions
)
Underfull \hbox (badness 10000) in paragraph at lines 39--41

 []


Underfull \hbox (badness 10000) in paragraph at lines 42--45

 []


Underfull \hbox (badness 10000) in paragraph at lines 46--49

 []

\myBox1=\box32
\myScoreStart1=\dimen113
\myCenter1=\dimen114
\myScoreEnd1=\dimen115
[1



{/usr/share/texmf-var/fonts/map/pdftex/updmap/pdftex.map}]
\myBox2=\box33
\myScoreStart2=\dimen116
\myCenter2=\dimen117
\myScoreEnd2=\dimen118

Overfull \hbox (30.21127pt too wide) in paragraph at lines 161--162
 [] 
 []

[2]
Underfull \hbox (badness 10000) in paragraph at lines 241--247

 []

[3]
Underfull \hbox (badness 10000) in paragraph at lines 248--251

 []


Underfull \hbox (badness 10000) in paragraph at lines 252--255

 []


Overfull \hbox (50.29764pt too wide) detected at line 274
[]
 []

[4] (./hw18b.aux) ) 
Here is how much of TeX's memory you used:
 904 strings out of 94500
 9181 string characters out of 1176766
 60724 words of memory out of 1000000
 4108 multiletter control sequences out of 10000+50000
 13065 words of font info for 51 fonts, out of 500000 for 2000
 580 hyphenation exceptions out of 1000
 27i,10n,26p,238b,210s stack positions out of 1500i,500n,5000p,200000b,5000s
PDF statistics:
 46 PDF objects out of 300000
 0 named destinations out of 131072
 1 words of extra memory for PDF output out of 65536
</usr/share/texmf/fonts/type1/bluesky/cm/cmsy7.pfb></usr/sha
re/texmf/fonts/type1/bluesky/cm/cmr7.pfb></usr/share/texmf/fonts/type1/bluesky/
cm/cmsy8.pfb></usr/share/texmf/fonts/type1/bluesky/cm/cmr8.pfb></usr/share/texm
f/fonts/type1/bluesky/cm/cmti10.pfb></usr/share/texmf/fonts/type1/bluesky/cm/cm
sy10.pfb></usr/share/texmf/fonts/type1/bluesky/cm/cmmi10.pfb></usr/share/texmf/
fonts/type1/bluesky/cm/cmtt10.pfb></usr/share/texmf/fonts/type1/bluesky/cm/cmbx
12.pfb></usr/share/texmf/fonts/type1/bluesky/cm/cmr10.pfb></usr/share/texmf/fon
ts/type1/bluesky/cm/cmbx10.pfb>
Output written on hw18b.pdf (4 pages, 77208 bytes).
