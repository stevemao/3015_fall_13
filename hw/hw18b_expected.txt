*Type_inference> infer [] (Abs "f" (Abs "x" (Abs "y" (Ap (V "f") (Pair (V "x") (V "y"))))))
(((a X b) -> c) -> (a -> (b -> c)))
*Type_inference> infer [] (Abs "f" (Abs "p" (Spread (V "p") ("x","y") (Ap (Ap (V "f") (V "x"))  (V "y")))))
((a -> (b -> c)) -> ((a X b) -> c))
*Type_inference> infer [] (Abs "f" (Abs "p" (Spread (V "p") ("x","y") (Ap (Ap (V "f") (V "y"))  (V "x")))))
((a -> (b -> c)) -> ((b X a) -> c))
*Type_inference> infer [] (Abs "f" (Abs "z" (Abs "w" (Ap  (V "f") (Pair (V "z") (V "w"))))))
(((a X b) -> c) -> (a -> (b -> c)))
*Type_inference> infer [] (Abs "p" (Spread (V "p") ("x","y") (V "x")))
((a X b) -> a)
*Type_inference> infer [] (Abs "p" (Spread (V "p") ("x","y") (V "y")))
((a X b) -> b)
*Type_inference> infer [] (Abs "p" (Spread (V "p") ("x","y") (Pair (V "y")(V "x"))))
((a X b) -> (b X a))
